exts/coding_guidelines/guidelines_checks.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: MIT OR Apache-2.0
+# SPDX-FileCopyrightText: The Coding Guidelines Subcommittee Contributors
+
+from sphinx.errors import SphinxError
+from sphinx_needs.data import SphinxNeedsData
+import logging
+
+logger = logging.getLogger('sphinx')
+
+class IntegrityCheckError(SphinxError):
+    category = "Integrity Check Error"
+
+def validate_required_fields(app, env):
+    """
+    Validate the required fields defined in conf.py 
+    """
+    logger.debug("Validating required fields")
+    data = SphinxNeedsData(env)
+    needs = data.get_needs_view()
+
+    required_fields = app.config.required_guideline_fields  # Access the configured values
+
+    for key, value in needs.items():
+        if value.get('type') == 'guideline': # other types include rationale, non_compliant_example, compliant_example 
```

Comment:
I think we can remove this comment, should be okay.
```suggestion
        if value.get('type') == 'guideline':
```